platform/src/plot.rs

```diff
@@ -0,0 +1,81 @@
+use crate::config::Config;
+use plotters::coord::{Shift};
+use plotters::prelude::*;
+
+pub fn plot(config: &Config) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
`Box<dyn std::error::Error>` This just looks super fishy. Haven't dug into the code enough to understand, but it seems very abnormal.

Reply:
As a side question, can this evern return an error? I don't see any early returns or `?`s so I think it currently can't even fail. Shouldn't even need to return a result. Of course it is fine to for future proofing.

Reply:
Good point - in the second-to-last commit, some `?`s moved out of this function into child functions.

Reply:
Fixed in a new commit, thanks